android/app/src/main/java/com/epfl/dedis/hbt/ui/register/ScanPassportFragment.kt

```diff
@@ -0,0 +1,142 @@
+
+    private var scanner: TextRecognizer? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        // Create a request permission launcher which will ask for permission when launched
+        val requestPermissionLauncher = registerForActivityResult(
+            ActivityResultContracts.RequestPermission(),
+            requireActivity().activityResultRegistry
+        ) {
+            // This is the callback of the permission granter
+            applyPermissionToView()
+        }
+
+        _binding = FragmentPassportScanBinding.inflate(inflater, container, false).apply {
+            requestPermissionButton.setOnClickListener {
+                // Launch the permission request on click
+                requestPermissionLauncher.launch(
+                    Manifest.permission.CAMERA
+                )
+            }
+        }
+
+        return binding.root
+    }
+
+    override fun onResume() {
+        super.onResume()
+        applyPermissionToView()
+    }
+
+    override fun onDestroy() {
+        super.onDestroy()
+
+        scanner?.close()
+    }
+
+    private fun startCamera() {
+        val cameraController = LifecycleCameraController(requireContext())
+        val previewView: PreviewView = binding.previewView
+
+        scanner = TextRecognition.getClient(TextRecognizerOptions.DEFAULT_OPTIONS)
+
+        cameraController.setImageAnalysisAnalyzer(
+            ContextCompat.getMainExecutor(requireActivity()),
+            imageAnalyzerProvider.provide(
+                scanner!!,
+                CameraController.COORDINATE_SYSTEM_VIEW_REFERENCED,
+                ContextCompat.getMainExecutor(requireActivity())
+            ) {
+                val raw = it?.text ?: return@provide
+                val text = raw.replace(" ", "").replace("«", "<<")
+                val matcher1 = LINE_1_PATTERN.matcher(text)
+                val matcher2 = LINE_2_PATTERN.matcher(text)
+                if (matcher1.find() && matcher2.find()) {
+                    val names = matcher1.group()
+                        .substring(5)
+                        .split("<")
+                        .filter { s -> s.isNotEmpty() }
+                    MainActivity.setCurrentFragment(
+                        parentFragmentManager,
+                        RegisterFragment.newInstance(
+                            names.joinToString(" "),
+                            "",
+                            matcher2.group().substring(0, 9).replace("<", "")
+                        )
+                    )
+                }
+            }
+        )
+
+        cameraController.bindToLifecycle(this)
+        previewView.controller = cameraController
+    }
+
+    private fun applyPermissionToView() {
+        // Depending on the current permission state,
+        // show the preview or the request permission views
+        if (checkPermission()) {
```

Comment:
Please rename the function so that it is easier to read the code and understand immediately what the true or false values correspond to. For example, when a check() function returns true, it is ambiguous what it means, but when a isBlah() returns true, there is no question if it is Blah or not ;-)

```suggestion
        if (permissionIsGranted()) {
```